pxtcompiler/emitter/service.ts

```diff
@@ -1897,6 +1897,22 @@ namespace ts.pxtc.service {
 
             // check for explicit default in the attributes
             if (attrs && attrs.paramDefl && attrs.paramDefl[name]) {
+                if (typeNode.kind == SK.AnyKeyword) {
+                    const defaultName = attrs.paramDefl[name].toUpperCase();
+                    if (!Number.isNaN(+defaultName)) {
```

Comment:
```suggestion
                    if (!isNaN(defaultName)) {
```

isNaN coerces then checks, Number.isNaN does not coerce / behaves wonkily on non num input; for that reason I much prefer isNaN personally so that you don't get the "well it's not technically NaN" result (that is, `Number.isNaN("abc")` returns false, even though "abc" is not a number, which feels wrong / top level isNaN behaves as I would expect).

(if you / anyone else has a particular opinion against that I'd yield though, hah)

Reply:
oo okay so I just tried to do this and it's mad because defaultName isn't a number. is this not supposed to happen?

Reply:
Oh hmm, I didn't realize typescript had the typing for isNaN stricter than browsers behavior. I guess it doesn't make a difference in that case at all if you're forced to coerce, I'm used to using it from javascript code but no difference here then!